pkg/generators/models/testdata/cases/defaults/api.yaml

```
@@ -0,0 +1,176 @@
+openapi: 3.0.0
+info:
+  version: 0.1.0
+  title: Test
+
+paths: { }
+
+components:
+  schemas:
+
+    EnumString:
+      type: string
+      enum:
+        - foo
+        - bar
+      default: bar
+
+    EnumInteger:
+      type: integer
+      enum:
+        - 1
+        - 2
+        - 0
+      default: 2
+
+    ZeroInteger:
+      type: integer
+      format: int32
+      default: 0
+
+    EpsilonDouble:
+      type: number
+      format: double
+      default: 2.220446049250313e-16
+
+    PI:
+      type: number
+      format: double
+      default: 3.14159265359
+
+    Email:
+      type: string
+      format: email
+      default: "no@email.com"
+
+    String:
+      type: string
+      default: "a string"
+
+    Address:
+        type: object
+        required:
+          - street
+          - number
+          - city
+          - state
+        properties:
+          street:
+            type: string
+          number:
+            type: integer
+          city:
+            type: string
+          state:
+            type: string
+          zip:
+            type: string
+            nullable: true
+
+    Person:
+      type: object
+      required:
+        - name
+        - age
+        - address
+      properties:
+        name:
+          type: string
+        age:
+          type: integer
+          default: 13
+        address:
+          $ref: '#/components/schemas/Address'
+        pets:
+          type: array
+          items:
+            $ref: '#/components/schemas/Pet'
+
+    PetType:
+        type: string
+        enum:
+            - dog
+            - cat
+            - bird
+        default: dog
+
+    Pet:
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/PetType'
+        name:
+          type: string
+        age:
+          type: integer
+          default: 1
+      default:
+        type: dog
+        name: Fido
+        age: 2
+
+    Wrapper:
+      type: object
+      properties:
+        string:
+          $ref: '#/components/schemas/String'
+        email:
+          $ref: '#/components/schemas/Email'
+        enumString:
+          $ref: '#/components/schemas/EnumString'
+        enumInteger:
+          $ref: '#/components/schemas/EnumInteger'
+        person:
+          $ref: '#/components/schemas/Person'
+
+        integerArray:
+          type: array
+          items:
+            type: integer
+          default: [ 1, 3 ]
+
+        bool:
+          type: boolean
+          default: true
+
+        cat:
+          $ref: '#/components/schemas/Pet'
+
+        inlineStruct:
+          type: object
+          properties:
+            field1:
+              type: string
+            field2:
+              type: integer
+              default: 13
```

It might be interesting to have multiple nested inline structs, just to have a solid example of what happens with naming. This is hopefully a very rare case, but worth at least showing what happens. We actually forbid unnamed inline objects like this, because there is always a possibility for a weird name conflict.

```suggestion
        inlineStruct:
          type: object
          properties:
            field1:
              type: string
            field2:
              type: integer
              default: 13
            nestedInline:
              type: object
              properties:
                name:
                  type: string
                cost:
                  type: integer
                  default: 100
```